### PR TITLE
 #159730632 Configure storybook for component development

### PR DESCRIPTION
**Pull request by hariclerry**
#### What does this PR do?
This PR configures the storybook tool to make working on related components
 more efficient, easier to reuse and have better component documentation.

#### Description of Task to be completed?
- Install and configure storybook into the App
- Run storybook using Docker commands

#### How should this be manually tested?
- Clone the repository
- Install all dependencies using `yarn install`
- Ensure docker is installed, to install docker click [here](https://www.docker.com/products/docker-desktop)
- Run `make components` docker build command on the terminal to run storybook in the docker development environment server.
- On the browser, type the localhost URL ` http://localhost:9001` and you should be able to see the storybook page displayed.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#159730632](https://www.pivotaltracker.com/story/show/159730632)

#### Screenshots (if appropriate)

screen shot 2018-08-14 at 17 28 20

#### Questions:
None

### CODE REVIEW

File: .storybook/config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { configure } from '@storybook/react';
+
+function loadStories() {
+  require('./stories/index.js');
+  // You can require as many stories as you need.
+}
+
+configure(loadStories, module);
```
Comment (albert85) [inline .storybook/config.js]: Enter a new Line after the last line of code

File: .storybook/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+
+storiesOf('Button', module)
+  .add('with text', () => (
+    <button onClick={action('clicked')}>Hello Button</button>
+  ))
+  .add('with some emoji', () => (
+    <button onClick={action('clicked')}><span role="img" aria-label="so cool">😀 😎 👍 💯</span></button>
+  ));
```
Comment (albert85) [inline .storybook/stories/index.js]: Same here also

File: docker/dev/docker-compose.yml [hunk 3]
```diff
@@ -12,3 +12,16 @@ services:
...
+      dockerfile: docker/dev/Dockerfile
+    volumes:
+      - /usr/app/node_modules
+      - ../../:/usr/app
+    ports:
+      - '9001:9001'
+    command:
+      - yarn 
+      - run
+      - storybook
```
Comment (albert85) [inline docker/dev/docker-compose.yml]: Same here

File: README.md [hunk 4]
```diff
@@ -42,6 +42,12 @@ List of libraries, tools, etc needed (e.g. yarn, node.js, python, etc)
 
 - Run `make test` on the terminal to run test inside containers
 
+## To use Storybook
+
+- Ensure docker is installed, to install docker click [here](https://www.docker.com/products/docker-desktop)
+- Run `make start` docker build command to run storybook in the docker development environment
+- Start the server using `yarn run storybook`
```
Comment (samuelkubai) [inline README.md]: @hariclerry how is this expected to run? Where should we run `yarn run storybook`?
I thought we agreed we would have a `make components` command to launch storybook through docker.

Comment (hariclerry) [reply README.md]: @samuelkubai, I will edit that accordingly, about the make components I will reach out to @nyambati or @HarithJ and figure it out.

File: docker/dev/docker-compose.yml (hunk 3 above)
Comment (hariclerry) [reply docker/dev/docker-compose.yml]: Working on it.

File: .storybook/stories/index.js (hunk 2 above)
Comment (hariclerry) [reply .storybook/stories/index.js]: Working on it.

File: .storybook/config.js (hunk 1 above)
Comment (hariclerry) [reply .storybook/config.js]: Workin on it.

**Comment by hariclerry** [comment]
@albert85, I have implemented the requested changes.

**Comment by hariclerry** [comment]
@samuelkubai, I ahve implemented the feedback you suggested.

**Comment by hariclerry** [comment]
@samuelkubai, changes implemented.

**Pull request merged**